core/src/serve_repair.rs

```diff
@@ -1070,9 +1074,12 @@ impl ServeRepair {
             .unzip();
         let peers = WeightedShuffle::new("repair_request_ancestor_hashes", &weights)
             .shuffle(&mut rand::thread_rng())
-            .take(ANCESTOR_HASH_REPAIR_SAMPLE_SIZE)
             .map(|i| index[i])
-            .map(|i| (repair_peers[i].id, repair_peers[i].serve_repair))
+            .filter_map(|i| {
+                let addr = repair_peers[i].serve_repair().ok()?;
+                Some((repair_peers[i].id, addr))
+            })
+            .take(ANCESTOR_HASH_REPAIR_SAMPLE_SIZE)
```

Comment:
yeah, this is technically a no-op because `repair_peers` is already filtered out for the ones with valid `serve_repair` socket. but I did not want to introduce an `unwrap` here, so just wrote it this way.

Reply:
Makes sense - thank for clarifying